### PR TITLE
Email

### PR DESCRIPTION
**Pull request by JackyRahman**
### Debio 928, 866, 913
[https://blocksphere2020.atlassian.net/browse/DBIO-928](https://blocksphere2020.atlassian.net/browse/DBIO-928)
[https://blocksphere2020.atlassian.net/browse/DBIO-866](https://blocksphere2020.atlassian.net/browse/DBIO-866)
[https://blocksphere2020.atlassian.net/browse/DBIO-913](https://blocksphere2020.atlassian.net/browse/DBIO-913)

### Debio 928 API Reward

- add flow bounty

### Debio 866 Email

- create template for send email when lab register
- create fucntion for send email when lab register
- fixing template and fuction for send email when customer staking request service

### Debio 913 Lab indexer

- add query and filter for service flow
- add field verification
- update types.json

### CODE REVIEW

**Review by rubenkristian — APPROVED** [review]
looks good to me

File: src/mail/mail.service.ts [hunk 1]
```diff
@@ -32,11 +32,42 @@ export class MailService {
...
+    lab_name: string,
+    country: string,
+    state: string,
+    city: string,
+    address: string,
+    certificates: Array<any>,
+    services: Array<any>
+  ){
+    await this.mailerService.sendMail({
+      to: 'jackyrahman443@gmail.com',
```
Comment (agustinustheo) [inline src/mail/mail.service.ts:49]: Why are you hardcoding this?

File: src/mail/mail.service.ts [hunk 2]
```diff
@@ -32,11 +32,42 @@ export class MailService {
...
+        country,
+        state,
+        city,
+        address,
+        certificates,
+        services
+      }
+    })
+  }
+  // =======testing and how to use this function===========
```
Comment (agustinustheo) [inline src/mail/mail.service.ts:66]: Please remove unnecessary comments like these.

File: src/substrate/geneticTestingEvent.ts [hunk 3]
```diff
@@ -84,24 +96,37 @@ export default class GeneticTestingEventHandler implements OnModuleInit {
...
   }
 
-  onDnaSampleComputed(event) {
-    console.log('DnaSampleComputed TODO: handle event');
-  }
+  async onDataStaked(event) {
+    console.log('DataStaked');
+    const dataStaked = event.data[0].toJSON();
+    const dataOrder = await (
+      await this.api.query.orders.orders(dataStaked.order_id)
```
Comment (agustinustheo) [inline src/substrate/geneticTestingEvent.ts:112]: You should separate queries to the blockchain inside a `polkadot-providers` folder, please refer to frontend: https://github.com/debionetwork/debio-customer-ui/tree/development/src/common/lib/polkadot-provider

**Review by agustinustheo — CHANGES_REQUESTED** [review]

**Review by agustinustheo — APPROVED** [review]

**Pull request merged**